pkgs/top-level/aliases.nix

```diff
@@ -5,110 +5,134 @@ with self;
 let
   # Removing recurseForDerivation prevents derivations of aliased attribute
   # set to appear while listing all the packages available.
-  removeRecurseForDerivations = alias: with lib;
+  removeRecurseForDerivations = alias:
+    with lib;
     if alias.recurseForDerivations or false then
-      removeAttrs alias ["recurseForDerivations"]
-    else alias;
+      removeAttrs alias [ "recurseForDerivations" ]
+    else
+      alias;
 
   # Disabling distribution prevents top-level aliases for non-recursed package
   # sets from building on Hydra.
-  removeDistribute = alias: with lib;
-    if isDerivation alias then
-      dontDistribute alias
-    else alias;
+  removeDistribute = alias:
+    with lib;
+    if isDerivation alias then dontDistribute alias else alias;
 
   # Make sure that we are not shadowing something from
   # all-packages.nix.
-  checkInPkgs = n: alias: if builtins.hasAttr n super
-                          then throw "Alias ${n} is still in all-packages.nix"
-                          else alias;
+  checkInPkgs = n: alias:
+    if builtins.hasAttr n super then
+      throw "Alias ${n} is still in all-packages.nix"
+    else
+      alias;
 
   mapAliases = aliases:
-    lib.mapAttrs (n: alias: removeDistribute
-                             (removeRecurseForDerivations
-                              (checkInPkgs n alias)))
-                     aliases;
-in
+    lib.mapAttrs (n: alias:
+      # Check if the alias' value is a string from `removedReason`; this helps
+      # avoid confusing errors:
+      #     https://github.com/NixOS/nixpkgs/pull/101098
+      if lib.isString alias then
+        throw "Attribute ${n} in <nixpkgs> has been removed; ${alias}"
```

Comment:
`<nixpkgs>` might not always exist (e.g. in a flakes-based system). Maybe just `s/<nixpkgs>/Nixpkgs/`?